src/components/SinglePayment.js

```diff
@@ -1,16 +1,23 @@
 import React from "react";
 import Button from "./Button";
 
-const SinglePayment = ({ payment }) => {
+const SinglePayment = ({ payment, index, setPaymentData, paymentData, allPaymentData }) => {
   const { date, currency, amount, description, status } = payment;
+
+  const cancelPendingPayment = () => {
+    setPaymentData([...allPaymentData.slice(index + 1)]);
```

Comment:
`slice` method returns a new array so [...] should not be needed.

TODO: On a side note, instead of removing one item here, we want to be changing the status to `complete`. Remember you still need to care for referential equality here